### ISSUE TITLE
FR: Add ACL mechanism to allow access to approved subnet routes of a machine or tag

### ISSUE THREAD

**Issue by xaverkapeller** [tailscale/tailscale#12832]
### What are you trying to do?

td;dr: I want to allow access to a specific subnet without knowing the exact subnet CIDR in advance. 

In our environment the subnet CIDRs are randomly created when a node boots. We have no way to influence this or restrict it to some known or predefined subnet.

Each of our nodes are assigned ACL tags and the advertised subnet routes are automatically approved by an application which verifies that everything is in order.

if I now want to allow access to these subnets I have to either:
- Dynamically update ACL with the exact IPs and subnet CIDRs (which we really don't want to do)
- Use wildcard rules e.g. `*:*` in our ACL (which we also really don'y want to do)

### How should we solve this?

If there was some globbing mechanism in the ACL syntax to refer to approved routes of a node this problem would be solved. To illustrate what I mean:

```json
    {
      "action": "accept",
      "src":    ["tag:some-tag"],
      "dst": [
        "tag:another-tag:subnet-routes",
        "100.101.68.118:subnet-routes",
        "some-user@example.com:subnet-routes",
      ],
    },
```

In this example the `:subnet-routes` after either a tag, tailscale IP or user would identify the subnet routes advertised by the machine and subsequently allow access to these subnet routes from `tag:some-tag`.

Not sure if this is the best way to implement a feature like this in the ACL, but I think you get the picture.

### What is the impact of not solving this?

Without a feature like this we would have to either dynamically update ACL through the API whenever a new node boots or create wildcard ACL rules.

Updating the ACL through the API is something we'd like to avoid because it makes the ACL far too clunky and unwieldy to have potentially hundreds entries for random IPs and CIDRs in it, especially so since the ACL is just a single json. In addition, a bug in the application that updates the ACL could break our networking.

At the moment we have worked around the issue by creating wildcard rules for specific ports e.g. `*:6443`. This solution works, but it's by far not perfect. if we could "glob" the subnet routes of specific tags or machines, it would allow us to restrict access as much as possible, while keeping the ACL simple and concise.

**Comment by woohgit** [comment]
We kind of need exactly the same approach.

With the `autogroup:internet`  - example below - we allow everyone to use subnet routers / exit nodes but what if we want to allow a different subset of users to able to access different subnet router advertised routes.


```
{
    "action": "accept",
    "src":    ["autogroup:member"],
     "dst":    ["autogroup:internet:*"],
},
```

This FR would make fine-grained subnet router / exit node work instead of letting everyone or nobody.

**Comment by tkcontiant** [comment]
Yep, this is exactly what I am trying to achieve.

Group: dev to access only subnets advertised by a particular tag-subnet-router:*


Example policy:

```json
{
	"groups": {
		"group:dev": [
			"member1@domain",
			"member2@domain",
		],

		"group:production": [
			"member_prod1@domain",
			"member_prod_2@domain",
		],
	},

	"acls": [
		{
			"action": "accept",
			"src": [
				"group:dev",
			],
			"dst": [
				"tag:k8s-vpc-infra-connector:*",
				"autogroup:member:*",
			],
		}
	],
}
```

**Comment by notturingtested** [comment]
We would like the same thing as well. Our cloud provider changes the subnets, and we use the operator to create a connector to them, but current functionality means that we still have to go fix this manually.

**Comment by CCob** [comment]
Yup, this is exactly the issue I came up against too. Would love to see this feature

**Comment by democai** [comment]
Definitely want some shorthand for "Allow any route that this tag provides".  Without it the process is manual and brittle.

**Comment by robertsLando** [comment]
+1 for this, please consider adding this feature
